### ISSUE TITLE
One-to-Many relation display blank screen if mainField is empty

### ISSUE THREAD

**Issue by opgbaudouin** [strapi/strapi#1230]
**Informations**
- **Node.js version**:9.5.0
- **npm version**:5.6.0
- **Strapi version**:3.0.0-alpha.12.1.3

- **Database**:mongodb 3.6
- **Operating system**:windows 10 x64

**What is the current behavior?**
I created a new content type called 'License'
I created a new content type called 'Item'
I created a relationship (One to Many) between a License and a User (from the User-Permissions)
So a User can have many Licenses.
The 'fields' are called:
User.licenses (-> [ Licenses ] )
License.licensee  (-> Single User)
As such: 
![user_has_many_licenses](https://user-images.githubusercontent.com/19164973/40247360-09c4d684-5ac4-11e8-85df-7adad1f577e5.jpg)

There is also a relationship between an Item and License (Item.licenses -> ( [Licenses]) )

I create a new a new License (either via REST API or via admin interface) and i can see the License easily

![license_ok](https://user-images.githubusercontent.com/19164973/40247404-2868db08-5ac4-11e8-8ccc-162284babce6.jpg)

However when i go to the user itself and just want to edit i get a totally blank screen (it flashes for a second) and i get a 2 errors:

```
err TypeError: Cannot read property 'id' of undefined
    at main.js:32
    at Array.map (<anonymous>)
    at t.value (main.js:32)
    at p (main.js:16)
    at beginWork (main.js:16)
    at a (main.js:16)
    at i (main.js:16)
    at b (main.js:16)
    at v (main.js:16)
    at y (main.js:16)
```
Obviously the UI is crashing... 

The license is kept in mongodb as such:

```
{
"_id":"5aff00a47376ac0fd881ebc5",
"createdAt":"2018-05-18T16:34:44.749Z",
"updatedAt":"2018-05-18T16:34:44.751Z",
"__v":0,
"item":"5af2ba83c6b8fb6d28d76291",
"id":"5aff00a47376ac0fd881ebc5",   //Other License objects which have licensee==null do NOT have this.
"licensee":"5af1a6cf23253c3588080503"
}
```
However the 'user'  doesn't contain anything with Licensee - nor can i see anything else in mongodb where this would be kept (not an issue as it is one-to-many)

My User.settings.json (the added bit)
```
  ...
  "licences": {
      "collection": "license",
      "dominant": true,
      "via": "licensee"
    },
    "items": {     
      "collection": "item",
      "via": "creator",
      "dominant": true
    },
```
and my License.settings.json:
```
  "licensee": {
      "model": "user",
      "via": "licences",
      "plugin": "users-permissions"
    },
```
**Steps to reproduce the problem**

See above

**What is the expected behavior?**

1) for the UI to show something (at least) - i.e. this is unhandled exception so very hard to debug.
2) when 1) to show the list of licenses that this user has. 

**Suggested solutions**

I'm out of my depth - as just learning..

---

- [x] I'm sure that this feature hasn't already been referenced.

**Comment by indiecore-fm** [comment]
anytime I create a one to many relation or many to one relation strapi crashes with this error:
`[2018-05-20T00:45:13.994Z] error Something went wrong in the model `Series` with the attribute `subSeries`
[2018-05-20T00:45:13.996Z] error Cannot read property 'attributes' of undefined`

**Comment by maxtor3569** [comment]
Same error, set mainField fix it for me ;)

**Comment by opgbaudouin** [comment]
Further info: this happens all across the place when i think React is expecting text type and it getting something else (or empty strings). If i put a mainField to a text and leave it empty - it also completely blanks...

**Comment by opgbaudouin** [comment]
And more: when updating my license in the admin UI, it will wipe the 'mainField' from the settings.json. So it appears the mainField (though critical for admin UI work) is not yet fully integrated.

**Comment by lauriejim** [comment]
@opgbaudouin can you try with the last version of Strapi `3.0.0-alpha.12.2` actually. Check on new app or [update your application](https://github.com/strapi/strapi/wiki/migration-guide-alpha.12.1-to-alpha.12.2).

**Comment by opgbaudouin** [comment]
Still same issue on 3.0.0-alpha.12.2 

Only works with mainField present to a 'meaningfull' - text based part.

**Comment by lauriejim** [comment]
Did you try with a fresh new app and new database ? Can you did it if it's not the case please.

I'm not able to reproduce your issue on `3.0.0-alpha.12.2`.

Can you share a .zip file of your `api` folder. And your User.settings.json file.

**Comment by opgbaudouin** [comment]
Did a clean install on my Centos 7 box with 3.0.0-alpha.12.2 

Issue is still there but my report was not accurate. 


Setup
==========

Create a type 'Item' 
  - content (single)
  - title (text) 

Create a type 'License'
  - pdf (single)
  - title (text) 

ADD to user type:
  - (User.)licenses <> (Users has and belongs to many Licenses) (License.)licensee
  - (User.)items <> (Users has and belongs to many Items) (Item.)creator

Add to type Item:
  - (Item.)licenses  < (Item belongs to many Licenses) (License.)item

Create an Item:
  - upload content
  - leave title empty

Create a license:
  - upload a pdf (or whatever)
  - leave the title empty

At this point it all works. 

Now EDIT your main (only) user if starting from scratch:
you will see a Licenses dropdown (it will show just white) put pick the 'first' (and only one).  It will show an ID. 

Press save

Now you are back in the overview. 

Try to edit the user again: 

Result -> Blank screen with same issues as before. main.js:16 TypeError: Cannot read property 'id' of undefined

However if you goto the Licenses / Edit only license you will see that the licensee is correct. If you remove it there you once again can edit the user. 

(Note this also happens when you try to link the User with an Item) - 

Now EDIT the Item and License and put actual texts in the title field. So they are no longer blank. 

Any relationship dropdown now has some non-blank text AND they no longer show 'id' text when selected. 

Therefore this issue might be more about 'Empty' title fields, or however strapi decides (besides mainField) to set the dropdown menus... 


User.settings.json:

```
   ... all the usual stuff from users.
     "licenses": {
       "collection": "license",
       "dominant": true,
       "via": "licensee"
     },
     "items": {
       "collection": "item",
       "dominant": true,
       "via": "creator"
     }
```
Item.settings.json:
```
{
  "connection": "default",
  "collectionName": "item",
  "info": {
    "name": "item",
    "description": "an item"
  },
  "options": {
    "timestamps": true
  },
  "attributes": {
    "licenses": {
      "collection": "license",
      "via": "item"
    },
    "content": {
      "model": "file",
      "via": "related",
      "plugin": "upload"
    },
    "creator": {
      "collection": "user",
      "via": "items",
      "plugin": "users-permissions"
    },
    "title": {
      "type": "string"
    }
  }
}
```
And License.settings.json:
```
{
  "connection": "default",
  "collectionName": "license",
  "info": {
    "name": "license",
    "description": "a License"
  },
  "options": {
    "timestamps": true
  },
  "attributes": {
    "license_pdf": {
      "model": "file",
      "via": "related",
      "plugin": "upload"
    },
    "item": {
      "model": "item",
      "via": "licenses"
    },
    "licensee": {
      "collection": "user",
      "via": "licenses",
      "plugin": "users-permissions"
    },
    "title": {
      "type": "string"
    }
  }
}
```

**Comment by opgbaudouin** [comment]
Apologies - i do not know 'Close' with close the item...

**Comment by lauriejim** [comment]
Hey @opgbaudouin thank you for this clear description !  

> Best steps description I ever seen. Thank you !

**Comment by lauriejim** [comment]
This issue is fixed in `3.0.0-alpha.14.1.1`
